lib/afc/index.js

```diff
@@ -0,0 +1,237 @@
+    fileModePayload.writeUInt32LE(fileHande, 0);
+
+    const data = {
+      opCode: Operations.FILE_CLOSE,
+      packetNumber,
+      headerPayload: fileModePayload
+    };
+    this._encoder.write(data);
+    const res = await response;
+    this.checkStatus(res);
+  }
+
+  async fileWrite (fileHande, buffer) {
+    const {packetNumber, response} = this.createPacketPromise();
+
+    const headerPayload = Buffer.alloc(8);
+    headerPayload.writeUInt32LE(fileHande, 0);
+
+    const data = {
+      opCode: Operations.FILE_WRITE,
+      packetNumber,
+      headerPayload,
+      content: buffer
+    };
+    this._encoder.write(data);
+    const res = await response;
+    this.checkStatus(res);
+  }
+
+  async fileRead (fileHande, length) {
+    const {packetNumber, response} = this.createPacketPromise();
+
+    const headerPayload = Buffer.alloc(16);
+    headerPayload.writeUInt32LE(fileHande, 0);
+    headerPayload.writeUInt32LE(length, 8);
+
+    const data = {
+      opCode: Operations.FILE_READ,
+      packetNumber,
+      headerPayload
+    };
+    this._encoder.write(data);
+    const res = await response;
+    if (res.opCode !== Operations.DATA) {
+      this.checkStatus(res);
+    }
+    return res.content;
+  }
+
+  async getFileInfo (path) {
+    const {packetNumber, response} = this.createPacketPromise();
+
+    const data = {
+      opCode: Operations.GET_FILE_INFO,
+      packetNumber,
+      headerPayload: Buffer.from(path)
+    };
+    this._encoder.write(data);
+    const res = await response;
+    if (res.opCode !== Operations.DATA) {
+      this.checkStatus(res);
+    }
+    return this.parseObject(res.content);
+  }
+
+  checkStatus (res) {
+    if (res.opCode !== Operations.STATUS) {
+      throw new Error(`Unexpected response ${operationCode(res.opCode)}`);
+    }
+    if (_.isEmpty(res.headerPayload)) {
+      throw new Error('Header payload cant be empty for a status response');
+    }
+    if (res.headerPayload[0] !== Errors.SUCCESS) {
+      throw new Error(`Unexpected response ${errorCode(res.headerPayload[0])}`);
+    }
+  }
+
+  parseArray (buffer) {
+    const items = [];
+    let start = 0;
+    for (let end = 0; end < buffer.length; end++) {
+      if (buffer[end] !== NULL_DELIMETER_CODE) {
+        continue;
+      }
+      const item = buffer.toString('utf8', start, end);
+      items.push(item);
+      // We skip the null delimeter
+      start = end + 1;
+    }
+    return items;
+  }
+
+  parseObject (buffer) {
+    const items = {};
+    let start = 0;
+    let currentKey;
+    for (let end = 0; end < buffer.length; end++) {
+      if (buffer[end] !== NULL_DELIMETER_CODE) {
+        continue;
+      }
+      const item = buffer.toString('utf8', start, end);
+      if (_.isNil(currentKey)) {
+        currentKey = item;
+      } else {
+        items[currentKey] = item;
+        currentKey = null;
+      }
+      // We skip the null delimeter
+      start = end + 1;
+    }
+    return items;
```

Comment:
what if currentKey has a value here, but it is not written into the resulting object?

Reply:
I can throw an error there to be defensive